include/flux/core/inline_sequence_base.hpp

```diff
@@ -7,6 +7,7 @@
 #define FLUX_CORE_INLINE_SEQUENCE_BASE_HPP_INCLUDED
 
 #include <flux/core/sequence_access.hpp>
+#include <flux/core/simple_sequence_base.hpp>
```

Comment:
If we're going to re-use it in other places, I'd prefer to move the `optional_like` concept into `core/concepts.hpp` (but keep it in a `detail` namespace for now) and remove this `#include`